fp/utils.py

```diff
@@ -0,0 +1,43 @@
+import pandas as pd
+
+def filter_optimal_results_skyline_order(_df, _order_list):
+    _df['selection_rate'] = abs(1 - _df['selection_rate'])
+    higher_is_better = ['num_true_positives', 'num_true_negatives', 'num_generalized_true_positives',
+                        'num_generalized_true_negatives', 'true_positive_rate', 'true_negative_rate',
+                        'generalized_true_positive_rate', 'generalized_true_negative_rate', 'positive_predictive_value',
+                        'accuracy', 'num_pred_positives']
+    lower_is_better = ['selection_rate', 'num_false_positives', 'num_false_negatives',
+                       'num_generalized_false_positives', 'num_generalized_false_negatives', 'false_positive_rate',
+                       'false_negative_rate', 'generalized_false_positive_rate', 'generalized_false_negative_rate',
+                       'false_discovery_rate', 'false_omission_rate', 'negative_predictive_value', 'error_rate',
+                       'num_pred_negatives']
+    order = []
+    for item in _order_list:
+        if item in higher_is_better:
+            order.append(False)
+        else:
+            order.append(True)
+    _df = _df.sort_values(_order_list, ascending=order)
+
+    return _df.values[0]
+
+
+def filter_optimal_results_skyline_formula(_df, _formula):
```

Comment:
please give names to variables that explain what data they contain, using just df and _df is very confusing

Reply:
Fixed by deleting it. This file should be deleted for the refactored code.